relay_messages.go

```diff
@@ -91,13 +91,22 @@ func (cr lazyCallRes) OK() bool {
 	return cr.Payload[_resCodeIndex] == _resCodeOK
 }
 
+type keyVal struct {
+	key []byte
+	val []byte
+}
+
 type lazyCallReq struct {
 	*Frame
 
 	caller, method, delegate, key, as []byte
 
+	headerStartOffset              int
+	headerEndOffset                int
```

Comment:
looking at the code, this refers to the transport headers. If we need to keep these, let's add that to the name, so it's clear this isn't arg2 headers.

It looks like we use this when creating a new frame? Not 100% clear why these are needed, when we need to create a new frame, shouldn't we just copy the frame up to `arg2`?
 *

Reply:
the checksum changes when arg2 is modified (which comes right after the headers) so we can't just blindly copy over everything, but will have to
- write header
- defer checksum
- write arg1 with checksum.Add(arg1)
- write arg2 with checksum.Add(arg2)
- write arg3 with checksum.Add(arg3)
- checksumRef.Update(checksum.Sum())

will rename to transportHeaders

Reply:
Why not:
 * Copy frame up to arg2
 * Store offset of where checksum is located
 * checksum.Add(arg1)
 * Write arg2 and arg3, while also updating checksum
 * Modify checksum using the offset of where checksum is located